### PR TITLE
Feature/task 5085 adapt and migrate database

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
This PR is about :
Adding new fields and update funcs to the contact journal store.
Adding migration logic for the new database schema.
Implementing corresponding tests.
Moving tests of 1To2 migration to own test class.
Implementing thread safe logic for serial migration on the database queue.

## Link to Jira
[5085](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5085)

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStore.swift [hunk 1]
```diff
@@ -505,15 +504,109 @@ class ContactDiaryStore: DiaryStoring, DiaryProviding {
...
+		circumstances: String
+	) -> DiaryStoringVoidResult {
+		var result: DiaryStoringVoidResult?
+
+		databaseQueue.inDatabase { database in
+			Log.info("[ContactDiaryStore] Update ContactPersonEncounter with id: \(id).", log: .localData)
+
+			let sql = """
+				UPDATE ContactPersonEncounter
+				SET date = SUBSTR(?, 1, 250), duration = ?, maskSituation = ?, setting = ?, circumstances = SUBSTR(?, 1, 250)
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStore.swift:522]: I personally would have handled that in code not SQL. E.g. changing this limit requires to look up for all occurrences of the '250' value. Just mentioning.

Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStore.swift]: Good point. I have added a constant for the max text length.

**Review by tcfos — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/StoreV3/ContactDiaryStoreSchemaV3.swift [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+	@discardableResult
+	func create() -> Result<Void, SQLiteErrorCode> {
+		var result: Result<Void, SQLiteErrorCode> = .success(())
+
+		databaseQueue.inDatabase { database in
+			let sql = """
+				CREATE TABLE IF NOT EXISTS ContactPerson (
+					id INTEGER PRIMARY KEY,
+					name TEXT NOT NULL CHECK (LENGTH(name) <= 250),
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/StoreV3/ContactDiaryStoreSchemaV3.swift:26]: Could also use the new constant for maxTextLength

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/__tests__/ContactDiaryStoreTests.swift [hunk 3]
```diff
@@ -137,77 +226,163 @@ class ContactDiaryStoreTests: XCTestCase {
...
+
+		let result = store.addLocationVisit(
+			locationId: locationId,
+			date: "2020-12-10",
+			durationInMinutes: 42,
+			circumstances: "Some circumstances."
+		)
+
+		guard case let .success(locationVisitId) = result
+		else {
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/__tests__/ContactDiaryStoreTests.swift:278]:
```suggestion
		guard case let .success(locationVisitId) = result else {
```

**Review by nickguendling — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStore.swift [hunk 4]
```diff
@@ -20,7 +20,7 @@ struct DateProvider: DateProviding {
 // swiftlint:disable:next type_body_length
 class ContactDiaryStore: DiaryStoring, DiaryProviding {
 
-	static let encryptionKey = "ContactDiaryStoreEncryptionKey"
+	static let encryptionKeyKey = "ContactDiaryStoreEncryptionKey"
```
[the remark this answers is not in the document]
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStore.swift:23]: Its the key for the encription key. The encryption key is stored in userdefaults under this key. 🙈

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/StoreV3/ContactDiaryStoreSchemaV3.swift (hunk 2 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/StoreV3/ContactDiaryStoreSchemaV3.swift:26]: I have added a property maxTextLength to the scheme.

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/__tests__/ContactDiaryStoreTests.swift (hunk 3 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/__tests__/ContactDiaryStoreTests.swift:278]: done

**Pull request merged**